File: hw/nase/u2/hw14 DBMongoandGUI/Python/model/appointment_model.py
from model.mongo_connection import MongoConnection

class AppointmentModel:
    def __init__(self):
        self.db = MongoConnection.get_database()
        self.collection = self.db["appointments"]

    def save_appointment(self, appointment_data):
        document = {
            "appointmentDate": appointment_data["appointmentDate"],
            "recommendations": appointment_data["recommendations"],
            "motherId": appointment_data["motherId"]
        }
        return self.collection.insert_one(document)

    def get_appointment_by_mother(self, mother_id):
        return self.collection.find_one({"motherId": mother_id}, sort=[("_id", -1)])

File: hw/nase/u2/hw14 DBMongoandGUI/JavaScript/model/appointmentModel.js
const MongoConnection = require('./mongoConnection');

class AppointmentModel {
    constructor() {
        this.db = MongoConnection.getDatabase();
        this.collection = this.db.collection("appointments");
    }

    async saveAppointment(appointmentData) {
        const document = {
            appointmentDate: appointmentData.appointmentDate,
            recommendations: appointmentData.recommendations,
            motherId: appointmentData.motherId
        };
        return await this.collection.insertOne(document);
    }

    async getAppointmentByMother(motherId) {
        return await this.collection.findOne({ motherId: motherId }, { sort: { _id: -1 } });
    }
}

module.exports = AppointmentModel;